pkg/security/ptracer/ptracer_test.go

```diff
@@ -0,0 +1,140 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+// Package ptracer holds ptracer related files
+package ptracer
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/kraken-hpc/go-fork"
+	"github.com/stretchr/testify/assert"
+)
+
+const fifoPath = "/tmp/.gotestfifo"
+
+func init() {
+	// register child for following fork
+	fork.RegisterFunc("child", child)
+	fork.Init()
+
+	// create fifo to communicate between parent and child
+	if _, err := os.Stat(fifoPath); os.IsNotExist(err) {
+		err := syscall.Mkfifo(fifoPath, 0666)
+		if err != nil {
+			fmt.Println("Error creating named pipe:", err)
+			os.Exit(1)
+		}
+	}
+}
+
+func child() {
+	// Open the FIFO for writing
+	fifo, err := os.OpenFile(fifoPath, os.O_WRONLY, os.ModeNamedPipe)
+	if err != nil {
+		os.Exit(1)
+		return
+	}
+	defer fifo.Close()
+
+	// Send PID to parent
+	binary.Write(fifo, binary.LittleEndian, int32(os.Getpid()))
+
+	// Child process: Listen for signals and notify parent
+	sigs := make(chan os.Signal, 1)
+	defer close(sigs)
+	signal.Notify(sigs)
+	defer signal.Reset()
+	for sig := range sigs {
+		_, err := fifo.Write([]byte(fmt.Sprintf("%v", sig)))
+		if err != nil {
+			os.Exit(1)
+		}
+	}
+	os.Exit(0)
+}
+
+func TestSignalForwarding(t *testing.T) {
+	forwardedSignals := []syscall.Signal{
+		syscall.SIGHUP,
```

Comment:
could we use a variable for the list of signal ? so that we keep the it in sync between test and startSignalForwarder

Reply:
good idea, applied on last commit